packages/dashboard/src/components/DataDesignerModal.tsx

```diff
@@ -64,9 +64,25 @@ export const DataDesignerModal: React.FC<DataDesignerModalProps> = ({ vizKey, ro
     return newData
   }
 
+  const removeDataset = (widgetKey = vizKey) => {
+    const newConfigureData = widgetKey ? config.visualizations[widgetKey] : config.rows[rowIndex]
+    delete newConfigureData.data
+    delete newConfigureData.dataKey
+    delete newConfigureData.dataDescription
+    delete newConfigureData.formattedData
+
+    return (newConfigureData as ConfigureData) || {}

```

Comment:
this function is directly manipulating the state. To accept ConfigureData and output ConfigureData